### PR TITLE
Allow filtering by funder on school comparison pages

### PR DESCRIPTION
**Pull request by ldodds**
Adds a new admin-only option on the "Choose country" tab of the school comparison page.

The new select box and filter allows an admin to further refine the list of schools to just those associated with a specific funder.

As the admins need to also narrow by country and school type it seemed better to add this as a new option rather than a new tab.

### CODE REVIEW

File: spec/system/compare_spec.rb [hunk 1]
```diff
@@ -390,6 +401,65 @@
...
+        it_behaves_like "a benchmark list page"
+        it_behaves_like "a filter summary", country: "Scotland", school_types_excluding: ['middle']
+        it_behaves_like "a filter summary", funder: "Grant Funder"
+
+        context "Changing options" do
+          before { click_on "Change options" }
+
+          it_behaves_like "an index page", tab: 'Choose country', show_your_group_tab: false
+          it_behaves_like "a form filter", id: '#country', country: 'scotland', school_types_excluding: ['middle']
+          it_behaves_like "a form filter", id: '#country', funder: "Funded by Grant Funder"
```
Comment (urbanwide) [inline spec/system/compare_spec.rb:441]: Should the id for this be '#funder'?

**Review by urbanwide — APPROVED** [review]
Looks good @ldodds - just one small comment about a test

File: spec/system/compare_spec.rb (hunk 1 above)
Comment (ldodds) [reply spec/system/compare_spec.rb:440]: It's checking within the countries tab (`#country`) for the expected state of the funder select box, so this is correct (and passes).

Comment (urbanwide) [reply spec/system/compare_spec.rb:440]: Sorry yes I can see that now 👍 Was a bit of a swift review, apologies

**Pull request merged**